packages/loopring_v3/contracts/impl/ExchangeDeployer.sol

```diff
@@ -24,26 +24,17 @@ import "./Exchange.sol";
 /// @author Daniel Wang  - <daniel@loopring.org>
 /// @dev We created this library to work around the gas limit -- inlining all the
 ///      enclosed function directly into LoopringV3 will make LoopringV3 too large
-///      to deploy.
+///      to deploy. Even if the deployed instance is initialized, it can still be
+///      used it is placed behind a proxy, therefore, he state (data) of the deployed
+///      instance is irrelevant when proxies are used.
 library ExchangeDeployer
 {
-    function deployExchange(
-        uint    exchangeId,
-        address payable loopringAddress,
-        address owner,
-        address payable operator,
-        bool    onchainDataAvailability
-        )
+    // Deploy an IExchange instance without initialization.
+
+    function deployExchange()
         external
         returns (address)
     {
-        Exchange exchange = new Exchange(
-            exchangeId,
-            loopringAddress,
-            owner,
-            operator,
-            onchainDataAvailability
-        );
-        return address(exchange);
+        return address(new Exchange());
```

Comment:
I’m thinking to initialize this instance in a way that nobody can use it as a standalone instance.

Reply:
Do we even need the ExchangeDeployer this any more? Because this deployment is only needed just once for every Exchange version it doesn't seem to make any sense to do it via ExchangeDeployer which also needs to be updated and redeployed for each Exchange version it deploys.

But it's possible someone could use the Exchange directly, yes. But if we deploy it we can disable it immediately by e.g. having an extra flag in the constructor of the Exchange which disables everything. There's probably some mechanism already out there that can be used for this.